File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/fragments/LocationFragment.java
package de.dimedis.mobileentry.fragments;

import static de.dimedis.mobileentry.util.ConfigPrefHelper.getBorders;
import static de.dimedis.mobileentry.util.ConfigPrefHelper.setUsersBorder;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;

import java.util.ArrayList;
import java.util.List;

import de.dimedis.mobileentry.adapter.ItemLocationAdapter;
import de.dimedis.mobileentry.backend.response.Border;
import de.dimedis.mobileentry.databinding.FragmentLocationBinding;
import de.dimedis.mobileentry.ui.ScanActivity;

public class LocationFragment extends BaseFragment {
    static final String TAG = "LanguageFragment";

    FragmentLocationBinding binding;

    public LocationFragment() {
    }

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        public Builder() {
        }

        public LocationFragment build() {
            return new LocationFragment();
        }
    }

    List<ItemLocationAdapter.Data> mapLanguage;


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);

        binding = FragmentLocationBinding.inflate(inflater, container, false);

        ItemLocationAdapter adapter = new ItemLocationAdapter(getLanguages());
        adapter.setOnItemClick((holder, data, position) -> {
            setUsersBorder(data.code);
            ScanActivity.scanEntry(getContext());
            getActivity().finish();
        });
        binding.location.setHasFixedSize(true);
        binding.location.setAdapter(adapter);
        binding.location.setLayoutManager(new LinearLayoutManager(getActivity()));

        return binding.getRoot();
    }

    List<ItemLocationAdapter.Data> getLanguages() {
        mapLanguage = new ArrayList<>();
        List<Border> borders = getBorders();
        if (null != borders) {
            for (Border entry : borders) {
                Log.i(TAG, "LANG " + entry.getBorderName() + "|" + entry.getFairName());
                mapLanguage.add(new ItemLocationAdapter.Data(entry));
            }
        }
        return mapLanguage;
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/fragments/LocationFragment.kt
package de.dimedis.mobileentry.fragments

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import de.dimedis.mobileentry.adapter.ItemLocationAdapter
import de.dimedis.mobileentry.databinding.FragmentLocationBinding
import de.dimedis.mobileentry.ui.ScanActivity
import de.dimedis.mobileentry.util.ConfigPrefHelper
import de.dimedis.mobileentry.util.ConfigPrefHelper.setUsersBorder

class LocationFragment : BaseFragment() {
    var binding: FragmentLocationBinding? = null

    class Builder {
        fun build(): LocationFragment {
            return LocationFragment()
        }
    }

    lateinit var mapLanguage: MutableList<ItemLocationAdapter.Data>

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        super.onCreateView(inflater, container, savedInstanceState)
        binding = FragmentLocationBinding.inflate(inflater, container, false)
        val adapter = ItemLocationAdapter(getLanguages())
        object : ItemLocationAdapter.OnItemClick {
            override fun onClick(holder: ItemLocationAdapter.ViewHolder, dat: ItemLocationAdapter.Data, position: Int) {
                setUsersBorder(dat.code)
                ScanActivity.scanEntry(requireContext())
                requireActivity().finish()
            }
        }

        binding!!.location.setHasFixedSize(true)
        binding!!.location.adapter = adapter
        binding!!.location.layoutManager = LinearLayoutManager(activity)
        return binding!!.root
    }

    fun getLanguages(): List<ItemLocationAdapter.Data> {
        mapLanguage = ArrayList()
        val borders = ConfigPrefHelper.getBorders()
        for (entry in borders) {
            Log.i(TAG, "LANG " + entry.borderName + "|" + entry.fairName)
            mapLanguage.add(ItemLocationAdapter.Data(entry))
        }
        return mapLanguage
    }

    companion object {
        const val TAG = "LanguageFragment"
        fun builder(): Builder {
            return Builder()
        }
    }
}